projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/KubernetesServiceTest.java

```diff
@@ -444,7 +442,15 @@ public void testCreateV1CronJobFromInternalResource() {
 
   @Test
   public void testReadJobDeploymentStatuses() {
-    KubernetesService mock = Mockito.mock(KubernetesService.class);
+    var mock =
+        Mockito.spy(
+            new DataJobsKubernetesService(
+                "default",
+                true,
+                new ApiClient(),
+                new BatchV1Api(),
+                new BatchV1beta1Api(),
+                new JobCommandProvider()));
```

Comment:
why not use the newDataJobKubernetesService function? I noticed that the only difference in the returned object is the boolean, maybe you could refactor that method to accept a param, and default to false so it doesn't require changing the other instances of the method call. Maybe this is overthinking it though.

Reply:
> why not use the newDataJobKubernetesService function? I noticed that the only difference in the returned object is the boolean

Yeah your description is spot on. 

Yeah I thought it was overkill because this is the only place we would pass in the parameter as true.

Reply:
gotcha, that sounds fine then